automapper.go

```diff
@@ -142,65 +142,83 @@ func mapFields(sourceVal, destVal reflect.Value, opts MapOptions) {
 func mapDestField(source, destVal reflect.Value, i int, opts MapOptions) {
 	destType := destVal.Type()
 	destTypeField := destType.Field(i)
-	fieldName := destTypeField.Name
+	destFieldName := destTypeField.Name
+	sourceFieldName := destFieldName
```

Comment:
Is it correct that you want to set the `sourceFieldName` as `destFieldName` here?